### PR TITLE
Issue #3364059 by zanvidmar: Add access check for entity queries

### PR DESCRIPTION
**Pull request by zanvidmar**
Add access check for entity queries used in social_group_flexible_group_update_11701 hook.

## Problem
social_group_flexible_group_update_11701 fails on database update.

> [error] 'status' not found
> [error] Update failed: social_group_flexible_group_update_11701
> [error] Update aborted by: social_group_flexible_group_update_11701
> [error] Finished performing updates.

Related to: https://www.drupal.org/project/social/issues/3348199

## Solution
[Access checking must be explicitly specified on content entity queries](https://www.drupal.org/node/3201242)

## Issue tracker
[3364059](https://www.drupal.org/project/social/issues/3364059)

## Theme issue tracker
N/A

## How to test
- [ ] Use PHP 7.4
- [ ] Create clean install of Open Social 10.3.9
- [ ] Update Open Social to 1.7.x-dev with composer
- [ ] Trigger database update (drush updb) and you will see "[error] 'status' not found", "[error] Update failed: social_group_flexible_group_update_11701"
- [ ] Database update is successful when repeating the steps with this fix applied.

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
N/A

## Release notes
Resolved an issue where the database update (social_group_flexible_group_update_11701) would encounter an error.

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3414-ib98sijsnn5fscvtlcxekttld4os43um.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/647862a16d7a7f9ed55f6934

File: modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install [hunk 1]
```diff
@@ -549,13 +549,14 @@ function social_group_flexible_group_update_11402(): string {
 function social_group_flexible_group_update_11701(array &$sandbox): string {
   if (!isset($sandbox['total'])) {
     $sandbox['total'] = \Drupal::entityQuery('group')
+      ->accessCheck(FALSE)
```
Comment (tbsiqueira) [inline modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install:552]: Are we sure that the access check should be FALSE? Default is TRUE. Same for below

**Review by tbsiqueira — CHANGES_REQUESTED** [review]

File: modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install (hunk 1 above)
Comment (zanvidmar) [reply modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install:552]: We want to update all the groups and I do not want that any access check alter interfere with process. However, you you might have an example why we should consider to changing this to TRUE? :)

Comment (tbsiqueira) [reply modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install:552]: Not really, just making sure we got the entire scenario covered.

**Review by tbsiqueira — APPROVED** [review]

File: modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install (hunk 1 above)
Comment (tbsiqueira) [reply modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.install:552]: Not really, just making sure we got the entire scenario covered.

**Pull request merged**